src/wagtailtrans/middleware.py

```diff
@@ -14,18 +14,36 @@
 class TranslationMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
+        active_language = None
         language_from_path = translation.get_language_from_path(request.path)
+        requested_languages = request.META.get('HTTP_ACCEPT_LANGUAGE')
         if language_from_path:
             active_language = language_from_path
-        elif get_wagtailtrans_setting('LANGUAGES_PER_SITE') and request.site:
-            site_languages = SiteLanguages.for_site(request.site)
-            active_language = site_languages.default_language.code
-        else:
-            default_language = Language.objects.default()
-            if default_language:
-                active_language = default_language.code
+        elif requested_languages:
+            requested_languages = requested_languages.split(',')
+            languages = Language.objects.filter(live=True)
+            codes = tuple([lang.code for lang in languages])
```

Comment:
for Language objects, we've implemented a manager which implements ``.live()`` for this I think it's better to use this, also using ``.values_list()`` makes it more readable

```codes = tuple(Language.objects.live().values_list('code', flat=True))```